### PR TITLE
Initial implementation of Fluo Docker image

### PR DESCRIPTION
**Pull request by mikewalch**
* Depends on local fluo-1.2.0-SNAPSHOT tarball

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,123 @@
...
+
+        docker run -d --network="host" mikewalch/fluo oracle -a myapp -D fluo.connection.zookeepers=zkhost/fluo
+
+2. Start Fluo worker(s). Execute this command multiple times to start multiple workers.
+
+        docker run -d --network="host" mikewalch/fluo worker -a myapp -D fluo.connection.zookeepers=zkhost/fluo
+
+## Marathon
+
+Using the Marathon UI, you can create applications using JSON configuration.
```
Comment (keith-turner) [inline README.md]: could link to marathon

File: README.md [hunk 2]
```diff
@@ -1 +1,123 @@
...
+        docker build -t fluo .
+
+   Or build the Fluo docker image with specific versions of Hadoop, Zookeeper, etc using the command below:
+
+        docker build --build-arg ZOOKEEPER_VERSION=3.4.8 --build-arg ACCUMULO_VERSION=1.7.3 --build-arg HADOOP_VERSION=2.7.0 -t fluo .
+
+## Image basics
+
+The entrypoint for the Fluo docker image is the `fluo` script. While the primary use
+case for this image is start an oracle or worker, you can run other commands in the
```
Comment (keith-turner) [inline README.md]: Need something after `is` in ` for this image is start an oracle`

File: README.md [hunk 3]
```diff
@@ -1 +1,123 @@
...
+While it is easier to pull from DockerHub, the image will default to the software versions below:
+
+| Software    | Version        |
+|-------------|----------------|
+| [Fluo]      | 1.2.0-SNAPSHOT |
+| [Accumulo]  | 1.8.1          |
+| [Hadoop]    | 2.7.3          |
+| [Zookeeper] | 3.4.9          |
+
+If these versions do not match what is running on your cluster, you should consider building
```
Comment (keith-turner) [inline README.md:19]: Would be nice to explicitly mention that this only works with 1.2+

File: README.md [hunk 4]
```diff
@@ -1 +1,123 @@
...
+your own image with matching versions. Below are instructions for building an image:
+
+1. Clone the Fluo docker repo
+
+        git clone git@github.com:mikewalch/fluo-docker.git
+
+2. Build a Fluo tarball distribution and copy it to the root directory of the repo.
```
Comment (keith-turner) [inline README.md]: could mention this step is only needed until 1.2.0 is released.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+ENV HADOOP_PREFIX /opt/hadoop
+ENV ZOOKEEPER_HOME /opt/zookeeper
+ENV ACCUMULO_HOME /opt/accumulo
+ENV FLUO_HOME /opt/fluo
+ENV PATH "$PATH:$FLUO_HOME/bin"
+
+RUN /opt/fluo/lib/fetch.sh extra
+
+ENTRYPOINT ["fluo"]
+CMD ["help"]
```
Comment (keith-turner) [inline Dockerfile:48]: what is this?  Its really nice how simple this dockerfile is.

File: README.md [hunk 6]
```diff
@@ -1 +1,123 @@
...
+```
+
+The JSON below can be used to start Fluo worker(s). Modify instances to start multiple workers.
+
+```json
+{
+  "id": "myapp-fluo-worker",
+  "cmd": "fluo worker -a myapp -D fluo.connection.zookeepers=zkhost/fluo",
+  "cpus": 1,
+  "mem": 128,
```
Comment (keith-turner) [inline README.md]: Should probably give more mem in this example config.

Do we need to open an issue about making Fluo dynamically adjust jvm settings to containers memory?

**Review by keith-turner — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (mikewalch) [reply README.md]: I increase memory in example and created #2

File: Dockerfile (hunk 5 above)
Comment (mikewalch) [reply Dockerfile:48]: when the docker container is run, the fluo script will be run by default with the "help" command. however, you can specify other commands.

**Pull request merged**